sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/ViewerApplication.java

```diff
@@ -0,0 +1,341 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.batch.protocol.viewer;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.util.Scanner;
+import javax.swing.JEditorPane;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSplitPane;
+import javax.swing.JTabbedPane;
+import javax.swing.JTree;
+import javax.swing.UIManager;
+import javax.swing.UIManager.LookAndFeelInfo;
+import javax.swing.event.TreeSelectionEvent;
+import javax.swing.event.TreeSelectionListener;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeModel;
+import javax.swing.tree.TreeSelectionModel;
+import org.sonar.batch.protocol.output.BatchReport;
+import org.sonar.batch.protocol.output.BatchReport.Component;
+import org.sonar.batch.protocol.output.BatchReport.Metadata;
+import org.sonar.batch.protocol.output.BatchReportReader;
+import org.sonar.batch.protocol.output.FileStructure.Domain;
+import org.sonar.core.util.CloseableIterator;
+
+public class ViewerApplication {
+
+  private JFrame frame;
+  private BatchReportReader reader;
+  private Metadata metadata;
+  private SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
+  private JTree componentTree;
+  private JSplitPane splitPane;
+  private JTabbedPane tabbedPane;
+  private JScrollPane treeScrollPane;
+  private JScrollPane componentDetailsTab;
+  private JScrollPane highlightingTab;
+  private JEditorPane componentEditor;
+  private JEditorPane highlightingEditor;
+  private JScrollPane sourceTab;
+  private JEditorPane sourceEditor;
+  private JScrollPane coverageTab;
+  private JEditorPane coverageEditor;
+  private TextLineNumber textLineNumber;
+
+  /**
+   * Launch the application.
+   */
+  public static void main(String[] args) {
+    EventQueue.invokeLater(new Runnable() {
+      public void run() {
+        try {
+          ViewerApplication window = new ViewerApplication();
+          window.frame.setVisible(true);
+
+          window.loadReport();
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+      }
+
+    });
+  }
+
+  private void loadReport() {
+    final JFileChooser fc = new JFileChooser();
+    fc.setDialogTitle("Choose batch report directory");
+    fc.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+    fc.setApproveButtonText("Open batch report");
+    int returnVal = fc.showOpenDialog(frame);
+    if (returnVal == JFileChooser.APPROVE_OPTION) {
+      File file = fc.getSelectedFile();
+      try {
+        loadReport(file);
+      } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)